### ISSUE TITLE
Add Scanner portal

### ISSUE THREAD

**Issue by hadess** [flatpak/xdg-desktop-portal#218]
This would require:
1. a portal, that asks for authorisation
1. a D-Busified version of saned, or saned itself with that feature, hopefully maintained upstream
1. a modified version of libsane that uses `"dbus"` instead of `"net"` as the driver, and accesses through the portal

2 and 3 would need discussion with the sane project, to hopefully have those features upstreamed, and enabled by default.

**Comment by AsciiWolf** [comment]
Any update?

**Comment by hadess** [comment]
> Any update?

This issue would have been updated if there were updates to be given, there's really no need to ask.

**Comment by lee-t-s** [comment]
Such a long time since this issue was raised 😮

**Comment by hadess** [comment]
> Such a long time since this issue was raised

And how does this comment help implement the feature exactly?

**Comment by nedrichards** [comment]
Indeed. As ever there's no lack of good ideas - just time and people committed to fixing them. Patches are *very* welcome.

**Comment by lee-t-s** [comment]
> > Such a long time since this issue was raised
> 
> And how does this comment help implement the feature exactly?

Perhaps the people who can help have not seen it till now.
I reached this bug by chance.

**Comment by PolyMeilex** [comment]
> 1. a portal, that asks for authorisation

I think we need to further discus what `authorisation` will actually mean.
I would split it in two separate topics: there is scanner authorization step (password for a scanner) and portal authorization.

So those are open questions that come to my mind:
1. Entering the password for a scanner could be done on portal side, that way the app would not know passwords to our devices?
2. Are we requiring a portal side scanner picker? Or do we just expose a list of devices and let the app do its thing? (aka. do we care about apps using device names for fingerprinting or whatever)
3. If we are exposing just a list, is it behind "Do you want to share your scanners with this app" kind of dialog, or is it just public?
4. In case we consider the list of devices to be public would the confirm dialog be more along the lines of "This app wants to access this specific scanner"

If we want to transparently integrate into libsane, we probably need a list of sorts instead of out of process picker. Unless we want to do a hack in which `sane_get_devices` always returns `["System Picker"]`

**Comment by bronson** [comment]
Just had a realization: personally I don't think I would find this useful anymore. Even if this feature lands, and Flatpak _could_ access SANE on the host, expecting the user to first make the proper edits to `/etc/sane.d/net.conf` is pretty steep. Not a great experience.

That said, I'm not quite sure how to fix it.  I suspect AirScan might be an answer? IIUC, the NAPS2 and InfiniScan Flatpaks bundle everything they need to use AirScan, regardless of how the host is configured. Install  the Flatpak, network scanning just works. If so, that would be a pretty good experience! But now I'll need to configure SANE to publish my USB scanner on the network so Flatpak'd apps can access it...  I'll have to research this more.
